.github/workflows/release.yml

```diff
@@ -0,0 +1,41 @@
+# This is a GitHub workflow to auto-deploy releases on tags
+# Useful references:
+# https://packaging.python.org/guides/publishing-package-distribution-releases-using-github-actions-ci-cd-workflows/
+# https://docs.github.com/en/free-pro-team@latest/actions/guides/building-and-testing-python#publishing-to-package-registries
+# https://github.community/t/how-to-run-github-actions-workflow-only-for-new-tags/16075
+# https://www.codingwithcalvin.net/git-tag-based-released-process-using-github-actions/
+name: Release
+
+# Always build releases (to make sure wheel-building works)
+# but only publish to PyPI on tags
+on:
+  push:
+  pull_request:
```

Comment:
https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet

I think the tags condition were interesting, but I think you got the regex syntax wrong in a recent commit.

Here is what I think would work and would be generally functional for all our repos in the jupyterhub org.

```
v?[0-9]+.[0-9]+.[0-9]+**
```

Reply:
I removed regex because gha was complaining about syntax :( Will try more to make it work if you too think it's interesting to have it (I liked it too when I saw it in a blog post :heart: )

> v?[0-9]+.[0-9]+.[0-9]+**

`v?`: nice  :+1:
`**`: according to the [cheat-sheet](https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet) (thanks for sharing it :heart:), one `*` won't match the `/` character, and `**` will match that. Do we want to match `/`? Is this commonly used in version names?

Reply:
I became uncertain and confused and thought that a single `*` would represent a single character only, but it turns out that it was matching many. I got mislead reading "The * wildcard matches any character, but does not match slash (/)." to think it was just a single character being matched.

So, then the question of `*` vs `**` is just a matter of inclusion of `/` or not, and I don't think we will ever experience a difference between these so choose either!